### DISCUSSION TITLE
Is it possible to configure postBuild to disable variable substitution by default?

### DISCUSSION THREAD

**Discussion by punkle** [fluxcd/flux2#1266] [Q&A] [answered]
Hi, we are using kustomization's postBuild feature to configure sub resources with variables. Some of our resources contain similar syntax embedded in string literals that we *do not* want substituted. We have the ability to go through the resources that have this problem and setting the substitute annotation to "false" and this works well for us.

However I think I would prefer if the substitute behavior would default to disabled and then enable it where necessary.

Is it possible to configure postBuild to disable variable substitution by default?

**Comment by stefanprodan** [comment, marked as answer]
> Some of our resources contain similar syntax embedded in string literals that we do not want substituted. 

From docs https://toolkit.fluxcd.io/components/kustomize/kustomization/#variable-substitution

> You can disable the variable substitution for certain resources by either labeling or annotating them with:
> `kustomize.toolkit.fluxcd.io/substitute: disabled`

**Reply by punkle** [reply]
@stefanprodan Is it possible to disable this by default on all resources and then use something like the following?

    kustomize.toolkit.fluxcd.io/substitute: enabled

**Reply by stefanprodan** [reply]
Nop but you can disable the whole Kustomization with `spec.suspend`.

**Reply by punkle** [reply]
Great thanks Stefan, that answers my question.

**Comment by eloo** [comment]
just came across this discussion and i'm wondering how `spec.suspend` is useful for this usecase...

i'm currently facing a similar issue where i want to deploy grafana alerts and dashboard with the same kustomization.
the kustomization is creating a configmap for alerts and one for dashboards.. for the alerts i want substitution but not for dashboards. 

yes i can disable it for dashboards and looks like this is working. but we have a couple of other kustomizations where at some point substitution would be useful but thats only a very small portion in this kustomization so configuring the variables and disable it by default would be great to have more control where substitution is going to happen and where not. 

so i would like to vote again for "disable by default" and "enable on specific kustomization/resource"

thanks

**Reply by samos667** [reply]
~~I'm facing exactly the same "issue" 
just some kind of escape character would be great~~

edit: my bad you can just $${someNonFluxVar}
